### PR TITLE
Issue 6214: Removed Spine nodes from gui

### PR DESCRIPTION
**Pull request by JCash**
The functionality has been moved to https://github.com/defold/extension-spine

### CODE REVIEW

File: engine/gui/src/dmsdk/gui/gui.h [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    /*#
+     * @name NewNode
+     * @param scene [type:HScene] the gui scene
+     * @param position [type:dmVMath::Point3] the position
+     * @param size [type:dmVMath::Vector3] the size
+     * @param node_type [type:dmGui::NodeType] the node type
+     * @param custom_type [type:uint32_t] If node_type == dmGui::NODE_TYPE_CUSTOM, then this is used to create a custom node data for the registered custom type
+     * @return node [type: dmGui::HNode] the created node
+     */
+    HNode NewNode(HScene scene, const dmVMath::Point3& position, const dmVMath::Vector3& size, NodeType node_type, uint32_t custom_type);
```
Comment (JCash) [inline engine/gui/src/dmsdk/gui/gui.h:171]: Although it's part of the public sdk due to extension-spine, we also probably don't want to promote it too much, as we want to remove the support for gui (in favour of equivalent functionality in the game object world)

File: engine/gui/src/gui.cpp [hunk 2]
```diff
@@ -406,8 +351,13 @@ namespace dmGui
...
         scene->m_FetchTextureSetAnimCallback = params->m_FetchTextureSetAnimCallback;
-        scene->m_FetchRigSceneDataCallback = params->m_FetchRigSceneDataCallback;
-        scene->m_RigEventDataCallback = params->m_RigEventDataCallback;
+        scene->m_CreateCustomNodeCallback = params->m_CreateCustomNodeCallback;
+        scene->m_DestroyCustomNodeCallback = params->m_DestroyCustomNodeCallback;
+        scene->m_CloneCustomNodeCallback = params->m_CloneCustomNodeCallback;
+        scene->m_UpdateCustomNodeCallback = params->m_UpdateCustomNodeCallback;
+        scene->m_CreateCustomNodeCallbackContext = params->m_CreateCustomNodeCallbackContext;
+        scene->m_GetResourceCallback = params->m_GetResourceCallback;
+        scene->m_GetResourceCallbackContext = params->m_GetResourceCallbackContext;
```
Comment (JCash) [inline engine/gui/src/gui.cpp:360]: Added new gui node hooks.

File: engine/gui/src/gui.h [hunk 3]
```diff
@@ -178,16 +153,20 @@ namespace dmGui
...
         dmParticle::HParticleContext m_ParticlefxContext;
         void*    m_UserData;
+        CreateCustomNodeCallback    m_CreateCustomNodeCallback;
+        DestroyCustomNodeCallback   m_DestroyCustomNodeCallback;
+        CloneCustomNodeCallback     m_CloneCustomNodeCallback;
+        UpdateCustomNodeCallback    m_UpdateCustomNodeCallback;
+        void*                       m_CreateCustomNodeCallbackContext;
+        GetResourceCallback         m_GetResourceCallback;
+        void*                       m_GetResourceCallbackContext;
+
```
Comment (JCash) [inline engine/gui/src/gui.h:168]: These are set by comp_gui.cpp

**Review by britzl — DISMISSED** [review]
There's a bunch of rig related code in comp_gui.cpp that has been commented out. Is there are reason why it's not removed?

**Comment by JCash** [comment]
Thanks, I'll remove that code! (I apparently reviewed the code too quickly)

**Pull request merged**